### ISSUE TITLE
IOU - IOU preview becomes more grayed out when creating another request to a pending request

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#34434]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.24-7
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Go offline.
2. Create a manual request with user with no prior chat.
3. Go to the main chat.
4. Create another manual request.

## Expected Result:
When creating another request, the IOU preview retains the same level of grayness.

## Actual Result:
When creating another request to a pending request in offline mode, the IOU preview becomes more grayed out.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/aaec4d18-8414-4e4d-9f5e-7c9a63be6b32

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by Tony-MK** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

### What is the root cause of that problem?

When the first money request is created offline, the bug does not occur because the [`ReportPreview`](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) component only dims when `iouReport.pendingFields.preview` is present. 

While the `props.action.pendingAction` will be set to `"add"`.

https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729

The `ReportActionItem` sets its `OfflineWithFeedback` component `pendingAction` prop to `props.action.pendingAction`. 

However, if `props.action.pendingAction` is not present and if `props.action.isOptimisticAction` is `true` then the `pendingAction` prop to will be set to `CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD`.

After another money requests are created offline, the `iouReport.pendingFields.creatChat` is unchanged, and `iouReport.pendingFields.preview` is added as `"update"`.

The `iouReport.pendingFields` will look like this.

![Screenshot 2024-01-14 at 05 11 31](https://github.com/Expensify/App/assets/22982173/be233d8b-19b9-4cac-b066-fa914bb32bc1)

Therefore, the root cause of the problem is the [`ReportPreview`](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) component sets its `OfflineWithFeedback` component to dimm, even if the component [`ReportActionItem`'s ](https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729) `OfflineWithFeedback` component is already dimming. 

Hence, the [`ReportPreview`](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) component will be double-dimmed.

This is why the [`ReportPreview`'s](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) `pendingAction` prop uses `iouReport.pendingFields.preview` rather than `iouReport.pendingFields.createChat`.

### What changes do you think we should make in order to solve the problem?


We can not disable dimming for [`ReportActionItem`](https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729) because when `isWhisper` is `true`, it will not dim its [child components](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/pages/home/report/ReportActionItem.js#L736-L75700) such as `DisplayNames`. 

This is because the [`ReportPreview`](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) component is the only component doing the dimming when the `props.action.actionName` is equal to `REPORTPREVIEW` and not [`ReportActionItem`](https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729).

Due to the previous fact and the only component that uses [`ReportPreview`](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) is the [`ReportActionItem`](https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729) component, I suggest we only do minor changes to the [`ReportPreview`](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) component. 

Hence, we should enable the `shouldDisableOpacity` prop to the `OfflineWithFeedback` child component of `ReportPreview` using a condition that will only be `true` if the `props.action` has a `pendingAction` or `isOptimisticAction`.

https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261

Therefore, the props for the `OfflineWithFeedback` in [`ReportPreview`](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) code snippet above should look like the code snippet below.

```
<OfflineWithFeedback 
      pendingAction={lodashGet(props, 'iouReport.pendingFields.preview')}> // No Change
      shouldDisableOpacity={props.action.pendingAction || props.action.isOptimisticAction} // Only change
>
```

**Result**
![Result](https://github.com/Expensify/App/assets/22982173/3586cff7-0752-4575-8a26-880c077f73cf)

**Comment by Tony-MK** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

IOU preview becomes more grayed out when creating another request to a pending request 

### What is the root cause of that problem?

When the first money request is created offline, the bug does not occur because the [`ReportPreview`](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) component only dims when `iouReport.pendingFields.preview` is present. 

While the `props.action.pendingAction` will be set to `"add"`.

https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729

The [`ReportActionItem`](https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729) sets its `OfflineWithFeedback` component `pendingAction` prop to `props.action.pendingAction`. 

However, if `props.action.pendingAction` is not present and if `props.action.isOptimisticAction` is `true` then the `pendingAction` prop to will be set to `CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD`.

After another money requests are created offline, the `iouReport.pendingFields.creatChat` is unchanged, and `iouReport.pendingFields.preview` is added as `"update"`.

The `iouReport.pendingFields` will look like this.

![Screenshot 2024-01-14 at 05 11 31](https://github.com/Expensify/App/assets/22982173/be233d8b-19b9-4cac-b066-fa914bb32bc1)

Therefore, the root cause of the problem is the [`ReportPreview`](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) component sets its `OfflineWithFeedback` component to dimm, even if the component [`ReportActionItem`'s ](https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729) `OfflineWithFeedback` component is already dimming. 

Hence, the [`ReportPreview`](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) component will be double-dimmed.

This is why the [`ReportPreview`'s](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) `pendingAction` prop uses `iouReport.pendingFields.preview` rather than `iouReport.pendingFields.createChat`.

### What changes do you think we should make in order to solve the problem?


We can not disable dimming for [`ReportActionItem`](https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729) because when `isWhisper` is `true`, it will not dim its [child components](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/pages/home/report/ReportActionItem.js#L736-L75700) such as `DisplayNames`. 

This is because the [`ReportPreview`](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) component is the only component doing the dimming when the `props.action.actionName` is equal to `REPORTPREVIEW` and not [`ReportActionItem`](https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729).

Due to the previous fact and the only component that uses [`ReportPreview`](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) is the [`ReportActionItem`](https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729) component, I suggest we only do minor changes to the [`ReportPreview`](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) component. 

Hence, we should enable the `shouldDisableOpacity` prop to the `OfflineWithFeedback` child component of `ReportPreview` using a condition that will only be `true` if the `props.action` has a `pendingAction` or `isOptimisticAction`.

https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261

Therefore, the props for the `OfflineWithFeedback` in [`ReportPreview`](https://github.com/Expensify/App/blob/fd45dea98390e74e7953a764a56c32c4da541a68/src/components/ReportActionItem/ReportPreview.js#L261) code snippet above should look like the code snippet below.

```
<OfflineWithFeedback 
      pendingAction={lodashGet(props, 'iouReport.pendingFields.preview')}> // No Change
      shouldDisableOpacity={props.action.pendingAction || props.action.isOptimisticAction} // Only change
>
```

**Result**
![Result](https://github.com/Expensify/App/assets/22982173/3586cff7-0752-4575-8a26-880c077f73cf)

**Comment by sonialiap** [comment]
@cubuspl42 what do you think of the above proposals?

**Comment by cubuspl42** [comment]
Both proposals agree that the right place to fix this issue is adjusting the properties passed `OfflineWithFeedback` (within `ReportPreview`): `pendingAction` and `shouldDisableOpacity`.

@Tony-MK Is there an observable difference when using the logic you suggest as compared to the proposal by @dukenv0307? Can we observe different dimming behavior?

**Comment by Tony-MK** [comment]
Thank you for reviewing, @cubuspl42.

My prospal aims to fix the fact that the part below won't be dimmed if we disable the opacity of the [ReportActionItem](https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729) if the component is a `ReportPreview`.

https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L736-L758

Therefore, the only the fix it is by using `shouldDisableOpacity` only at the `ReportPreview` component, not at the [ReportActionItem](https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729) component.

In conclusion, that is the main visual difference noticed when testing.

What do you think? 🤔

**Comment by Tony-MK** [comment]
Thank you for the review, @cubuspl42.

My prospal aims to fix the fact that the part below won't be dimmed if we disable the opacity of the [ReportActionItem](https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729) component if it is a `ReportPreview`.

https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L736-L758

Therefore, the only the fix it is by using `shouldDisableOpacity` only at the `ReportPreview` component, not at the [ReportActionItem](https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729) component.

> Both proposals agree that the right place to fix this issue is by adjusting the properties passed OfflineWithFeedback (within ReportPreview): pendingAction and shouldDisableOpacity.

My proposal only suggests modifying the `shouldDisableOpacity` for the `ReportPreview` component.

Unlike the other proposal which suggests changing the `shouldDisableOpacity` for the [ReportActionItem](https://github.com/Expensify/App/blob/9533df38782001586ae1d7e6b4c513984fdf94de/src/pages/home/report/ReportActionItem.js#L727-L729) and pendingAction for the `ReportPreview`.

In conclusion, that is the main visual difference noticed when testing.

What do you think? 🤔

**Comment by cubuspl42** [comment]
I think I wasn't clear. By "observable" I meant observable by a user; you started explaining the code.

That's what I want to understand: whether the differences between our two proposals are purely technical (code-oriented), or is there an observable dimming difference.

What I ask can be explained with non-code terms only and (perfectly!) screenshots. Unless the answer is "the solutions are observable equivalent", of course.

**Comment by Tony-MK** [comment]
My apologies, @cubuspl42. I should have realized you wanted screenshots.

While testing scenarios, I noticed this difference in dimming behavior with the avatar and name components of the `ReportActionItem`.

This scenario can be reproduced if you created a chat or workspace online and then created a money request offline.

This dimming behavior is different from the current staging environment.

### Staging
![Staging](https://github.com/Expensify/App/assets/22982173/90e863e8-bb72-4e0d-8e83-8aa43f7c8b10)

### My Changes
![My Changes](https://github.com/Expensify/App/assets/22982173/3775c0bb-933b-4c9d-b8bd-6134e0eb99cc)

### Other Changes
![Other Propsal](https://github.com/Expensify/App/assets/22982173/29dc96a6-1d95-422a-b5c2-9bcdb48374b5)

**Comment by cubuspl42** [comment]
C+ reviewed 🎀 👀 🎀

I approve the proposal by @Tony-MK.

It was the second in the timeline order, but it was closer to the proper solution and the proposal text has a clear structure with a visible effort to make it easily understandable.

**Comment by cubuspl42** [comment]
@dukenv0307 Thanks! We'll make sure to test all cases that we mentioned here.

Your proposal was also good. in this case, I decided to select based on the proposal structure and initial solution correctness (they first noticed that we need a conditional `shouldDisableOpacity` logic).

**Comment by cubuspl42** [comment]
You have the right to your opinion, but the opinion that one's proposal is better is typically "a default" one. As I see it, it's necessary to show what C+ _missed_ to argue for changing their decision (after it's already made), not just disagree. Disagreeing with non-selecting of one's proposal is also "a default".

I think this is on the edge of breaking the rules against excessive commenting if not already breaking it. It was just my feedback, let's move on at least in this one.

**Comment by cubuspl42** [comment]
@Tony-MK Let me know when the PR is ready

**Comment by Tony-MK** [comment]
@cubuspl42, the [PR](https://github.com/Expensify/App/pull/35218) is ready for review.
Thank you.

**Comment by Tony-MK** [comment]
Hey everyone, I think @MelvinBot has forgotten about this issue 🤣 because [PR](https://github.com/Expensify/App/pull/35218) was deployed to production two weeks ago.

Kindly, could we finish off the issue? 🤔

**Comment by srikarparsi** [comment]
Hey! Yeah, @cubuspl42 is there anything left to do with this issue or is this ready for payment?

**Comment by cubuspl42** [comment]
@srikarparsi I confirm, the PR was deployed 2 weeks ago. We're ready for payment 🙂

**Comment by srikarparsi** [comment]
Cool thanks! @sonialiap do you think you could help with paying @Tony-MK $500 for coding the PR and @cubuspl42 $500 for reviewing the PR whenever you get a chance?

**Comment by sonialiap** [comment]
Payment summary:
- @Tony-MK $500 for coding the PR - paid ✔️ 
- @cubuspl42 $500 for reviewing the PR - paid ✔️
